### PR TITLE
feat: js enabled by default for 1 click binding

### LINKED ISSUE #31110: [Task]: Set Default JSMode to be enabled for 1-click binding
### SubTasks

Currently, when users use 1-click binding with a Query in JSONForm they see the bindings instead of the query name in UI. We need to enable it for all the widgets. 

image

- [ ] JSON Form ( for APIs )
- [x] Table widget 
- [x] Select widget
- [x] Multi-Select widget

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
When one click binding feature is used to bind the data with the widgets, in that case in order to make the users understand the dynamic bindings, we are enabling the js mode by default for `TableV2`, `Select`, `Multiselect` for both apis and queries. This solves one area of learnability.

This is behind a feature flag `rollout_js_enabled_one_click_binding_enabled`.

#### PR fixes following issue(s)
Fixes #31110 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Enhanced MultiSelect, Select, and Table widgets to support dynamic property paths, improving their interaction with dynamic data sources.
	- Introduced a new public method `ChooseQuery(queryName: string)` to facilitate selecting a query by name in the OneClickBinding class.

### CODE REVIEW

**Comment by rohan-arthur** [comment]
tested this, LGTM

**Review by rohan-arthur — COMMENTED** [review]
Are we considering feature flag conditions for this change?

**Review by hetunandu — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/JSEnabledByDefaultExperiment_spec.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import { featureFlagIntercept } from "../../../../support/Objects/FeatureFlags";
+import * as _ from "../../../../support/Objects/ObjectsCore";
+import EditorNavigation, {
```
[the remark this answers is not in the document]
Comment (AmanAgarwal041) [reply app/client/cypress/e2e/Regression/ClientSide/OneClickBinding/JSEnabledByDefaultExperiment_spec.ts]: Done

**Review by hetunandu — APPROVED** [review]

**Pull request merged**